File: 3019-NumberOfChangingKeys/cpp/solution.cpp
class Solution {
public:
    int countKeyChanges(string s) {
        if (s.empty()) return 0;
        int count = 0;
        char first = tolower(s[0]);
        for (char i : s) {
            char current = tolower(i);
            if (current != first) {
                count++;
                first = current;
            }
        }
        return count;
    }
};

File: 3019-NumberOfChangingKeys/javascript/solution.js
// Last updated: 3/6/2026, 12:39:29 AM
/**
 * @param {string} s
 * @return {number}
 */
var countKeyChanges = function(s) {
    let count = 0;
    let first = s[0].toLowerCase();
    for (let i of s) {
        i = i.toLowerCase();
        if (i != first) {
            count++;
            first = i;
        }
    }
    return count;
};